frontend/test/viewer/viewer.sagas.spec.ts

```diff
@@ -0,0 +1,147 @@
+			const containers = times(3, () => containerMockFactory());
+			const federations  = times(3, () => federationMockFactory());
+			const baseContainers = containers.map(prepareMockBasecontainer); 
+			const containerStat = prepareMockStatsReply(containers[1]);
+			const containerOrFederationId = containers[1]._id;
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/federations`)
+				.reply(200, { federations });
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/containers`)
+				.reply(200, {containers: baseContainers});
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/containers/${containerOrFederationId}/stats`)
+				.reply(200, containerStat);
+				
+			await waitForActions(() => {
+				dispatch(ViewerActionsCreators.fetchData(teamspace, projectId, containerOrFederationId));
+			}, [ViewerActionsCreators.setFetching(false)]);
+
+			const container = selectContainerById(getState(), containerOrFederationId);
+
+			expect(container).toEqual(prepareSingleContainerData(baseContainers[1], containerStat));
+		});
+
+		it('should fetch the containers, the federations, the federation data and the containers data for a particular federation', async () => {
+			const containers = times(6, () => containerMockFactory());
+			const federations = times(3, () => federationMockFactory());
+			const baseFederations = federations.map(prepareMockBaseFederation);
+			const containersStats = containers.map(prepareMockStatsReply);
+			const baseContainers = containers.map(prepareMockBasecontainer);
+			const containersInState = baseContainers.map((base, index) => prepareSingleContainerData(base, containersStats[index]));
+			const theFederation = federations[1];
+			theFederation.containers = [ baseContainers[2]._id, baseContainers[0]._id];
+			const containerOrFederationId = theFederation._id;
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/federations`)
+				.reply(200, { federations: baseFederations });
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/containers`)
+				.reply(200, {containers: baseContainers});
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/federations/${containerOrFederationId}/stats`)
+				.reply(200, prepareMockFederationStatsReply(theFederation));
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/containers/${baseContainers[0]._id}/stats`)
+				.reply(200, containersStats[0]);
+
+			mockServer
+				.get(`/teamspaces/${teamspace}/projects/${projectId}/containers/${baseContainers[2]._id}/stats`)
+				.reply(200, containersStats[2]);			
+				
+			await waitForActions(() => {
+				dispatch(ViewerActionsCreators.fetchData(teamspace, projectId, containerOrFederationId));
+			}, [ViewerActionsCreators.setFetching(false)]);
+
+
+			const federation = selectFederationById(getState(), containerOrFederationId);
+			const containersId = [...federation.containers].sort();
```

Comment:
I'd probably call this containersIds. containersId akes it sond like there's just one Id in total